apps/server/src/app/cache/cache.service.ts

```diff
@@ -1,40 +1,85 @@
 import { Injectable } from "@nestjs/common";
-import { CacheRequestDto } from "@pezzo/client";
+import {
+  CacheRequestDto,
+  CacheRequestResult,
+  CachedRequest,
+} from "@pezzo/client";
 import { createHash } from "crypto";
 import stableStringify from "json-stable-stringify";
+import { RedisService } from "../redis/redis.service";
 
-export interface CachedRequestResult {
-  projectId: string;
-  data: object;
-}
+// 3 days
+const REQUEST_CACHE_TTL = 60 * 60 * 24 * 3;
 
 @Injectable()
 export class CacheService {
-  private cache: Map<string, any> = new Map();
+  constructor(private redisService: RedisService) {}
 
-  private generateHash(request: object): string {
+  private generateHash(request: Record<string, any>): string {
     const stringified = stableStringify(request);
     const hash = createHash("sha256").update(stringified).digest("hex");
     return hash;
   }
 
-  fetchRequest(request: object): CachedRequestResult | null {
+  async fetchRequest(
+    request: Record<string, any>
+  ): Promise<CachedRequest | null> {
     const hash = this.generateHash(request);
+    const result = await this.redisService.client.json.get(`request:${hash}`, {
+      path: ".",
+    });
 
-    if (this.cache.has(hash)) {
-      return this.cache.get(hash);
+    if (!result) {
+      return null;
     }
 
-    return null;
+    this.redisService.client
+      .multi()
+      .expire(`request:${hash}`, REQUEST_CACHE_TTL)
+      .expire(`meta:request:${hash}`, REQUEST_CACHE_TTL)
+      .exec();
+
+    const metadata = await this.redisService.client.hGetAll(
+      `meta:request:${hash}`
+    );
+
+    return {
+      response: (result as unknown as Record<string, any>).response,
+      metadata: {
+        projectId: metadata.projectId,
+        organizationId: metadata.organizationId,
+      },
+    };
   }
 
-  cacheRequest(dto: CacheRequestDto): string {
+  async cacheRequest(dto: CacheRequestDto): Promise<CacheRequestResult> {
     const { request, response } = dto;
     const hash = this.generateHash(request);
 
-    this.cache.set(hash, {
-      data: response,
-    });
-    return hash;
+    await Promise.all([
```

Comment:
consider naming functions and keys for better readability, `e.g const metaHash = meta:request:${hash}`
```
const saveMeta =      this.redisService.client
        .multi()
        .hSet(metaHash, "projectId", dto.projectId)
        .hSet(metaHash, "organizationId", dto.organizationId)
        .expire(metaHash`, REQUEST_CACHE_TTL)
        .exec()